libnmstate/nm/route.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import logging
+from operator import itemgetter
+from libnmstate.schema import LinuxRoute
+
+from libnmstate import iplib
+
+
+def get_running(client, get_ip_cfg_func, skip_networks):
+    ret = []
+    for active_connection in client.get_active_connections():
+        ip_cfg = get_ip_cfg_func(active_connection)
+        if not ip_cfg:
+            continue
+        iface_name = _get_iface_name(active_connection)
+        if not iface_name:
+            logging.warning(
+                'Got connection {} has not interface name'.format(
+                    active_connection.get_id()))
+            continue
+        skip_networks.extend(_get_skip_networks(ip_cfg))
+        table_id = -1
+        # TODO: Waiting libnm to provide route table id for running state
```

Comment:
A workarround could be to check it using iproute2 or check if it is in the config and copy it from there.
But it can be "fixed" in a separate PR.

As it stands now, it will not be useful for clients like VDSM which use the tables extensively.

Reply:
Fixed.